pkg/curp/curp_test.go

```diff
@@ -0,0 +1,206 @@
+package curp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	curppb "github.com/xline-kv/go-xline/api/gen/curp"
+	xlineapi "github.com/xline-kv/go-xline/api/gen/xline"
+	"github.com/xline-kv/go-xline/pkg/rpc"
+)
+
+func TestUnaryFetchClusters(t *testing.T) {
+	allMembers := map[serverId][]string{
+		0: {"127.0.0.1:48081"},
+		1: {"127.0.0.1:48082"},
+		2: {"127.0.0.1:48083"},
+		3: {"127.0.0.1:48084"},
+		4: {"127.0.0.1:48085"},
+	}
+	unaryConfig := newUnaryConfig(1*time.Second, 2*time.Second)
+	unary, err := newUnaryBuilder(allMembers, unaryConfig).build()
+	assert.Nil(t, err)
+	res, err := unary.fetchCluster()
+	assert.Nil(t, err)
+	fcRes := rpc.NewFetchClusterResponse(res)
+	mems := fcRes.IntoMembersAddrs()
+	assert.Equal(t, 5, len(mems))
+	assert.Equal(t, []string{"A0"}, mems[0])
+	assert.Equal(t, []string{"A1"}, mems[1])
+	assert.Equal(t, []string{"A2"}, mems[2])
+	assert.Equal(t, []string{"A3"}, mems[3])
+	assert.Equal(t, []string{"A4"}, mems[4])
+}
+
+func TestUnaryFetchClustersFailed(t *testing.T) {
```

Comment:
Why does the `TestUnaryFetchClustersFailed` do the same thing as the `TestUnaryFetchClusters`, but the former fails while the latter succeeds?

Reply:
Because the mock server gives a different response, the return value can be found in the mock server configuration.

Reply:
But giving different responses based on the order of requests is not good for maintenance. I've added an Issue to track this.